src/plugins/intel_gpu/src/runtime/layout.cpp

```diff
@@ -136,6 +136,18 @@ layout layout::convert_to_weights_layout(bool is_grouped) const {
     return layout{data_type, fmt, size};
 }
 
+layout layout::get_layout_with_dims() const {
```

Comment:
As I mentioned above, how about remove this API and let the operator == to compare according to the dynamism?

Reply:
I reverted this commit (https://github.com/andrew-k-park/openvino/commit/664175c19211e1ab4e0e297e8784a33d6cc32750) and move partial shape comparison logic to equality operator in layout.